spatialprofilingtoolbox/apiserver/app/main.py

```diff
@@ -231,6 +225,46 @@ async def request_cggnn_metrics(
     return query().get_cggnn_metrics(study)
 
 
+@app.get("/cggnn-importance-composition/")
+async def cggnn_importance_composition(
+    study: ValidStudy,
+    positive_marker: ValidChannelListPositives,
+    negative_marker: ValidChannelListNegatives,
+    cell_limit: int = 100,
+) -> PhenotypeCounts:
+    """For each specimen, return the fraction of important cells expressing a given phenotype."""
+    cells_selected = cast(set[int], query().get_important_cells(study, cell_limit))
+    return get_phenotype_counts(
+        positive_marker,
+        negative_marker,
+        study,
+        len(cells_selected),
+        cells_selected,
+    )
```

Comment:
One more thing. Since most studies will have no importance information, we should handle that case well here. If `len(cells_selected) == 0`, we should return something other than a `PhenotypeCounts` instance. Maybe `None`? Can FastAPI handle that?

Reply:
Currently there is no real error handling implemented. If we will start to return different return types to indicate exceptional cases I would like it to be part of a pattern of exception handling that fits with the autogenerated OpenAPI docs. So maybe we can postpone this.

Reply:
As for when errors happen anyway (handled or not), FastAPI is pretty smart about logging what happened and continuing to serve requests. So the current state is OK.